src/stream/src/telemetry.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::OnceLock;
+
+use risingwave_common::telemetry::report::report_event_common;
+use risingwave_pb::telemetry::{PbTelemetryConnectorDirection, PbTelemetryEventStage};
+
+const TELEMETRY_COMPUTE_REPORT_TYPE: &str = "compute";
+static COMPUTE_TELEMETRY_SESSION_ID: OnceLock<String> = OnceLock::new();
+static COMPUTE_TELEMETRY_TRACKING_ID: OnceLock<String> = OnceLock::new();
+
+pub fn set_compute_telemetry_tracking_id_and_session_id(tracking_id: String, session_id: String) {
```

Comment:
Yes, but it needs to pass a telemetry manager instance when calling. Sometimes we want to know whether a corner case got hit, passing a telemetry manager instance can be annoying and involves more code changes. 

I plan to simplify the func calling process, encouraging ENG to collect features usage at minimal complexity. Directly calling a func is the simplest way so I have to use a global variable to keep the tracking_id and session_id.

Reply:
How about making it a singleton, such as `GLOBAL_METRICS`?

https://github.com/risingwavelabs/risingwave/blob/da28570ab58385ca5850f659ed0f96e3d9843151/src/expr/core/src/expr/expr_udf.rs#L270-L271

Reply:
For `session_id`, it varies for different nodes, and each time one node restarts. Putting it as a singleton can break the behavior, resulting in not being aligned with the prev reports.

Reply:
I don't fully understand the points. :eyes:
1. First, why do we need these `get_set` defined in different modules, but not just in `common`? Different nodes will just call it at different places. (Like how we share system params for different nodes)
2. Regardless of what their values are, we just created `tracking_id` and `session_id` once in `start_telemetry_reporting`, so why aren't they singletons? Can you give an example how can we break it?

Reply:
> For `session_id`, it varies for different nodes, and each time one node restarts. Putting it as a singleton can break the behavior, resulting in not being aligned with the prev reports.

Didn't get it. 😕 What's the different between a bunch of global singleton variables v.s. a global singleton `TemetryManager`? I think they just differ in code structure

Reply:
* The tracking ID is unique and immutable across the entire cluster. I support implementing this as a global singleton.
* The session ID is updated whenever a node is created or restarted. In single-node mode, all nodes run within the same process. Using a singleton in this context might cause these session IDs to overlap, failing to reflect the true state of the cluster.

Reply:
Thanks for clarifying that the only problem you are concerned about is single-node mode.

But I'm not sure why do we need to use different session ID for different nodes in single-node mode. What's the purpose of `session_id`?

Reply:
It marks how long a node lives, providing an identifier to show what happens in one session. For now, we have no analysis on it, not sure whether we want it in the future. 

> But I'm not sure why do we need to use different session ID for different nodes in single-node mode.

According to the git history, the logic is impl before introducing single-node mode.
In the prev impl, the session id is generated inside the func. The nodes get different session ids in both single-node mode and distributed mode. I introduce `OnceLock` to align the behavior.

Reply:
+1 for removing `session_id` because we don't really have such a purpose now. 

If we are worrying about restarts/crashes, consider reporting the events in the system table `system_events`, where critical events such as cause of recovery are stored.

Reply:
Okay, the expected behavior is 

* removing the session id part in the events framework, but keep it in stats report, may remove it in future pr.
* impl the tracking id as a singleton in telemetry_common.